packages/react-router/lib/components.tsx

```diff
@@ -154,6 +155,14 @@ export function RouterProvider({
     [router, navigator, basename]
   );
 
+  const wasRouterPreviouslyInitialized = useRef(false);
+  const isRouterInitialized = router.state.initialized;
+  const routerReinitializing =
+    wasRouterPreviouslyInitialized && !isRouterInitialized;
```

Comment:
Drive-by comment - this looks like a typo?
```suggestion
    wasRouterPreviouslyInitialized.current && !isRouterInitialized;
```